platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java

```diff
@@ -95,11 +100,35 @@ public ShadowGraphSynchronizer(
         this.syncMetrics = syncMetrics;
         this.generationsSupplier = generationsSupplier;
         this.syncDone = syncDone;
-        this.eventHandler = eventHandler;
         this.fallenBehindManager = fallenBehindManager;
         this.executor = executor;
         this.sendRecInitBytes = sendRecInitBytes;
         this.executePreFetchTips = executePreFetchTips;
+        this.eventHandler = buildEventHandler(platformContext, eventHandler);
+    }
+
+    /**
+     * Construct the event handler for new events. If configured to do so, this handler will also hash events before
+     * passing them down the pipeline.
+     */
+    private Consumer<GossipEvent> buildEventHandler(
+            @NonNull final PlatformContext platformContext, @NonNull final Consumer<GossipEvent> rawEventHandler) {
+
+        final boolean hashOnGossipThreads = platformContext
+                .getConfiguration()
+                .getConfigData(SyncConfig.class)
+                .hashOnGossipThreads();
+
+        if (hashOnGossipThreads) {
```

Comment:
Note: we already have logic in place that skips the hashing of an event if it is already hashed. See https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventValidator.java